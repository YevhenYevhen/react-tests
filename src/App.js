// import { useEffect, useState } from "react";

// function App() {
//   const [data, setData] = useState(null);
//   const [toggle, setToggle] = useState(false);
//   const [value, setValue] = useState('');


//   const onClick = () => setToggle(toggle => !toggle);

//   useEffect(() => {
//     setTimeout(() => setData({}), 100);
//   }, [])

//   return <div>
//     <h1 data-testid='value-elem'>{value}</h1>
//     {toggle && <div data-testid='toggle-elem'>toggle</div>}
//     {data && <div style={{ color: 'red' }}>data</div>}
//     <h1>Hello, world!</h1>
//     <button data-testid='toggle-button' onClick={onClick}>Click</button>
//     <input value={value} onChange={e => setValue(e.target.value)} type="text" placeholder="input value..." />
//   </div>
// }

// export default App;

import React from 'react';
import Navbar from './components/Navbar/Navbar';
import AppRouter from './router/AppRouter';


const App = () => {
  return <>
    <Navbar/>
    <AppRouter />
  </>
};

export default App;