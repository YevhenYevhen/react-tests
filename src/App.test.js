import { render, screen, fireEvent } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import App from './App';


describe.skip('Test App', () => {
  test('renders the component', () => {
    render(<App />);
    const h1 = screen.getByText(/hello, world!/i);
    const button = screen.getByRole('button');
    const input = screen.getByPlaceholderText(/input value.../i);
    expect(h1).toBeInTheDocument();
    expect(button).toBeInTheDocument();
    expect(input).toBeInTheDocument();
    expect(input).toMatchSnapshot();
  });

  test('doesn\'t render a non-existent component', () => {
    render(<App />);
    const h1 = screen.queryByText(/hello, world!!!/i);
    expect(h1).toBeNull();
  })

  test('conditionally renders a component', async () => {
    render(<App />);
    const div = await screen.findByText(/data/i);
    expect(div).toBeInTheDocument();
    expect(div).toHaveStyle({ color: 'red' });
  })

  test('toggle button should control if the element is in the document', () => {
    render(<App />);
    const toggleButton = screen.getByTestId('toggle-button');
    expect(screen.queryByTestId('toggle-elem')).toBeNull();
    fireEvent.click(toggleButton);
    expect(screen.queryByTestId('toggle-elem')).toBeInTheDocument();
    fireEvent.click(toggleButton);
    expect(screen.queryByTestId('toggle-elem')).toBeNull();
  })

  test('input change should update the value of the element', () => {
    render(<App />);
    const input = screen.getByPlaceholderText(/input value.../i);
    expect(screen.queryByTestId('value-elem')).toContainHTML('');
    // only input event
    // fireEvent.input(input, {
    //   target: { value: 'hello' }
    // });
    //input, keydown, keyup, etc. events 
    userEvent.type(input, 'hello');
    expect(screen.queryByTestId('value-elem')).toContainHTML('hello');
  })
})