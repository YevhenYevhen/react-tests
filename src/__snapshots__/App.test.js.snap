// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test App renders the component 1`] = `
<input
  placeholder="input value..."
  type="text"
  value=""
/>
`;
